### PR TITLE
Full decoupling of Pattern Lab from source/ design system

### PR DESCRIPTION
**Pull request by illepic**
Removes the entire concept of "demo" folders. Instead,  folders within `apps/node-pl` simply include namespaces from the source design system.

Todo:

- [x] yeoman generator must be updated to write files to two separate locations
- [x] further cruft cleanup

# Changes

Prior to this PR, our design system held both the "pure" component *and* the demonstration of this component, like so:

```
source/
  default/
    _patterns/
      01-atoms/
        button/
           _button.twig    # Pure component
           _button.scss    # Pure component's styles
           index.js        # Pure component's dependencies and logic
           demo/           # Pattern Lab only displays twig/yml within demo/
              buttons.yml  # Provide data to twig
              buttons.twig # The twig file visible in the PL UI
              index.js     # Put the PL twig/yaml on the PL dependency chain
```

With this PR, we move the presentational components **out** of `source/default/` to `apps/node-pl`:

```
apps/
  node-pl/
    pattern-lab/
      _patterns/  # Pattern Lab only displays everything in here
        01-atoms/
          button/
            buttons.yml  # Provide data to twig
            buttons.twig # The twig file visible in the PL UI
            index.js     # Put the PL twig/yaml on the PL dependency chain
source/
  default/
    _patterns/
      01-atoms/
        button/
           _button.twig    # Pure component
           _button.scss    # Pure component's styles
           index.js        # Pure component's dependencies and logic
```

The yeoman generator (`npm run new`), now places the proper files in both these places.

### CODE REVIEW

File: apps/node-pl/pattern-lab/_patterns/00-protons-demo/borders.twig [hunk 1]
```diff
@@ -94,3 +94,5 @@
     </div>
   </div>
 </div>
+
```
Comment (dslemay) [inline apps/node-pl/pattern-lab/_patterns/00-protons-demo/borders.twig]: Are these intentionally added?

Comment (illepic) [reply apps/node-pl/pattern-lab/_patterns/00-protons-demo/borders.twig]: If you're asking about the empty line at the end, yep!

Comment (dslemay) [reply apps/node-pl/pattern-lab/_patterns/00-protons-demo/borders.twig]: Roger that. Wasn't used to seeing two empty lines at the end of the file.

Comment (illepic) [reply apps/node-pl/pattern-lab/_patterns/00-protons-demo/borders.twig]: Whoops.

File: source/default/_patterns/02-molecules/carousel/_carousel.twig [hunk 2]
```diff
@@ -50,3 +50,5 @@
     {% endblock controls %}
   </div>
 </div>
+
```
Comment (Brigette-Eckert) [inline source/default/_patterns/02-molecules/carousel/_carousel.twig]: Did you mean to add extra spaces at end of file here?

Comment (illepic) [reply source/default/_patterns/02-molecules/carousel/_carousel.twig]: Fixed!

**Comment by sghoweri** [comment]
OH HEY, THIS LOOKS FAMILIAR @illepic! 😂

**Pull request merged**